.travis.yml

```diff
@@ -24,17 +24,17 @@ before_install:
   - if [[ $TRAVIS_PYTHON_VERSION == 3* ]]; then 2to3 -nw pysal/ > /dev/null; fi
 
 install:
-  - conda install --yes numpy scipy nose pip
+  - conda install --yes pip
   - if [ "$PYSAL_PLUS" == true ]; then
-        echo 'plus testing'; conda install --yes --file requirements_plus.txt;
-        else conda install --yes --file requirements.txt; 
+        echo 'plus testing'; pip install .[plus];
+    else pip install .;
     fi;
-  - pip install -r travis.txt
+  - pip install .[dev]
 
```

Comment:
thanks for review @ljwolf - do you know if this would work:

```
  - if [ "$PYSAL_PLUS" == true ]; then
        echo 'plus testing'; conda install .[plus];
    else conda install --yes .;
    fi;
  - conda install --yes .[dev]
```

i.e. can conda install from a directory with setup.py and resolves its depdendencies? 

i tried it in a docker container with mini-conda but it didn't work
```
root@5e0d2f572eec:/opt/pysal# ls
CHANGELOG.txt  INSTALL.txt  LICENSE.txt  MANIFEST.in  Makefile	PySAL.egg-info	README.rst  README.txt	THANKS.txt  Vagrantfile  authors.txt  distribute_setup.py  doc	figs  lattice.dbf  pysal  rtd.txt  setup.cfg  setup.py	tools  venv
root@5e0d2f572eec:/opt/pysal# conda install -y .
Fetching package metadata: ....
Solving package specifications: .
Error:  Package missing in current linux-64 channels:
  - .

You can search for this package on anaconda.org with

    anaconda search -t conda .

You may need to install the anaconda-client command line client with

    conda install anaconda-client
root@5e0d2f572eec:/opt/pysal#
```

it would be a shame to introduce something that requires to maintain package lists in setup.py and in other files.

maybe, if conda can't install from a local directory with a setup.py file, we can go back to files with the required packages, and read the files into setup.py.

What do you think?